main.cpp

```diff
@@ -18,13 +18,22 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto Expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (Expr) {
+            auto lborder = Expr->getLParenLoc();
+            auto rborder = Expr->getRParenLoc();

```

Comment:
Please describe the difference between the use of `getRParenLoc()` and `getSubExprAsWritten()->getBeginLoc()` with examples.

Reply:
The first returns the location of the right bracket where the cast type is written.
The second is the location of the beginning of the expression that is being cast.
For example: `(int)var`.
The first function will return the location of right branch, the second one will return the location of "v".